File: SERVER_PHP/app/Models/Message.php
<?php

namespace App\Models;

use Illuminate\Support\Facades\Auth;
use Jenssegers\Mongodb\Eloquent\Model as Eloquent;

class Message extends Eloquent {

    const LIMIT = 1;

    protected $collection = 'messages';
    protected $connection = 'mongodb';
    
    public function getMessagesWaitingByUser( $userId = 0 ){

        return $this->where('user', '=' ,$userId )->get();
    }

    // public function getListMessageInChannel($channel = '', $linit = self::LIMIT){
    //     $messages = $this->where('channel', new \MongoDB\BSON\ObjectID($conv->_id))->get(['read', '_id'])->toArray();
    //     foreach( $messages as $mess ){

    //     }
    // }

    public function getMessageByChannelId( $channel = '', $linit = self::LIMIT ){
        return $this->raw(function ($collection) use ( $channel, $linit ) {
            return $collection->aggregate([
                [
                    '$match' => [
                        "channel" => [
                            '$eq' => new \MongoDB\BSON\ObjectID($channel)
                        ]
                    ]
                ],
                [
                    '$project' => [
                        'user' => 1,
                        'body' => 1,
                        'read' => 1,
                        'readAdmin' => 1,
                        'style' => 1,
                        'attachment' => 1,
                        'channel' => 1,
                        'createdAt' => 1,
                        'updatedAt' => 1,
                    ]
                ],
                [
                    '$sort' => [ '_id' => -1 ],
                ],
                [
                    '$limit' => $linit,
                ],
                [
                    '$sort' => [ '_id' => 1 ],
                ],
            ]);

        });
    }
}

File: bds/lib/models/message.dart
class Message {
  String id;
  bool read;
  bool readAdmin;
  List<dynamic> attachment;
  int user;
  String body;
  String channel;
  String style;
  DateTime createdAt;
  DateTime updatedAt;

  Message({
    this.id,
    this.read,
    this.readAdmin,
    this.attachment,
    this.user,
    this.body,
    this.channel,
    this.style,
    this.createdAt,
    this.updatedAt,
  }) : super();

  factory Message.fromJson(Map<String, dynamic> json) => Message(
        id: json["_id"],
        read: json["read"] ?? false,
        readAdmin: json["readAdmin"] ?? false,
        attachment: List<dynamic>.from(json["attachment"].map((att) => att)),
        user: json["user"],
        body: json["body"],
        channel: json["channel"],
        style: json["style"],
        createdAt: DateTime.parse(json["createdAt"]),
        updatedAt: json["updatedAt"] == null
            ? DateTime.parse(json["createdAt"])
            : DateTime.parse(json["updatedAt"]),
      );

  Map<String, dynamic> toJson() => {
        "_id": id,
        "read": read,
        "readAdmin": readAdmin,
        "attachment": List<dynamic>.from(attachment.map((att) => att)),
        "user": user,
        "body": body,
        "channel": channel,
        "style": style,
        "createdAt": createdAt.toIso8601String(),
        "updatedAt": updatedAt.toIso8601String(),
      };
}
